weave

```diff
@@ -888,20 +888,33 @@ ipam_reclaim() {
     done
 }
 
-# Call IPAM as necessary to allocate addresses
-# $1 is the full container id and following args are previously parsed CIDR_ARGS.
-# Returns ALL_CIDRS and IPAM_CIDRS
+# Call IPAM as necessary to lookup or allocate addresses
+#
+# $1 is one of 'lookup', 'allocate' or 'allocate_unchecked', $2 is the
+# full container id. The remaining args are previously parsed
+# CIDR_ARGS.
+#
+# Populates ALL_CIDRS and IPAM_CIDRS
 ipam_cidrs() {
-    CHECK_ALIVE="?check-alive=true"
-    if [ "$1" = "--no-check-alive" ] ; then
-        CHECK_ALIVE=
-        shift 1
-    fi
-    CONTAINER_ID="$1"
-    shift 1
+    case $1 in
+        lookup)
+            METHOD=GET
+            CHECK_ALIVE=
+            ;;
+        allocate)
+            METHOD=POST
+            CHECK_ALIVE="?check-alive=true"
+            ;;
+        allocate_unchecked)
```

Comment:
`unchecked` suggests "no check for error" to me.  How about `allocate_no_check_alive` ?